src/native/fs/fs_napi.cpp

```diff
@@ -1941,6 +1948,51 @@ struct DirOpen : public FSWorker
     }
 };
 
+struct TellDir : public FSWrapWorker<DirWrap>
+{
+    long int _tell_res = -1;
+    TellDir(const Napi::CallbackInfo& info)
+        : FSWrapWorker<DirWrap>(info)
+    {
+        Begin(XSTR() << "TellDir " << DVAL(_wrap->_path));
+    }
+    virtual void Work()
+    {
+        DIR* dir = _wrap->_dir;
```

Comment:
add a check like we do in DirReadEntry for the case that dir was closed - if (!dir) ...